engines/sci/graphics/video32.cpp

```diff
@@ -0,0 +1,390 @@
+		} else if (_stretchVertical) {
+			vmdScaleInfo.y = 256;
+			vmdScaleInfo.signal = kScaleSignalDoScaling32;
+			vmdRect.bottom += vmdRect.height();
+		}
+
+		const int16 screenWidth = g_sci->_gfxFrameout->getCurrentBuffer().screenWidth;
+		const int16 screenHeight = g_sci->_gfxFrameout->getCurrentBuffer().screenHeight;
+		const int16 scriptWidth = g_sci->_gfxFrameout->getCurrentBuffer().scriptWidth;
+		const int16 scriptHeight = g_sci->_gfxFrameout->getCurrentBuffer().scriptHeight;
+
+		BitmapResource vmdBitmap(_segMan, vmdRect.width(), vmdRect.height(), 255, 0, 0, screenWidth, screenHeight, 0, false);
+
+		if (screenWidth != scriptWidth || screenHeight != scriptHeight) {
+			mulru(vmdRect, Ratio(scriptWidth, screenWidth), Ratio(scriptHeight, screenHeight), 1);
+		}
+
+		CelInfo32 vmdCelInfo;
+		vmdCelInfo.bitmap = vmdBitmap.getObject();
+		_decoder->setSurfaceMemory(vmdBitmap.getPixels(), vmdBitmap.getWidth(), vmdBitmap.getHeight(), 1);
+
+		if (_planeIsOwned) {
+			_x = 0;
+			_y = 0;
+			_plane = new Plane(vmdRect, kPlanePicColored);
+			if (_priority) {
+				_plane->_priority = _priority;
+			}
+			g_sci->_gfxFrameout->addPlane(*_plane);
+			_screenItem = new ScreenItem(_plane->_object, vmdCelInfo, Common::Point(), vmdScaleInfo);
+		} else {
+			_screenItem = new ScreenItem(_plane->_object, vmdCelInfo, Common::Point(_x, _y), vmdScaleInfo);
+			if (_priority) {
+				_screenItem->_priority = _priority;
+			}
+		}
+
+		if (_blackLines) {
+			_screenItem->_drawBlackLines = true;
+		}
+
+		// NOTE: There was code for positioning the screen item using insetRect
+		// here, but none of the game scripts seem to use this functionality.
+
+		g_sci->_gfxFrameout->addScreenItem(*_screenItem);
+
+		_decoder->start();
+	}
+
+	EventFlags stopFlag = kEventFlagNone;
+	while (!g_engine->shouldQuit()) {
+		if (_decoder->endOfVideo()) {
+			stopFlag = kEventFlagEnd;
+			break;
+		}
+
+		g_sci->getEngineState()->speedThrottler(_decoder->getTimeToNextFrame());
+		g_sci->getEngineState()->_throttleTrigger = true;
+		if (_decoder->needsUpdate()) {
+			renderFrame();
+		}
+
+		const int currentFrameNo = _decoder->getCurFrame();
+
+		if (_yieldInterval > 0 &&
+			currentFrameNo != _lastYieldedFrameNo &&
+			(currentFrameNo % _yieldInterval) == 0
+		) {
+			_lastYieldedFrameNo = currentFrameNo;
+			stopFlag = kEventFlagYieldToVM;
+			break;
+		}
+
+		const SciEvent event = _eventMan->getSciEvent(SCI_EVENT_MOUSE_PRESS | SCI_EVENT_KEYBOARD);
+
+		if ((flags & kEventFlagMouseDown) && event.type == SCI_EVENT_MOUSE_PRESS) {
+			stopFlag = kEventFlagMouseDown;
+			break;
+		}
+
+		if ((flags & kEventFlagEscapeKey) && event.type == SCI_EVENT_KEYBOARD && event.character == SCI_KEY_ESC) {
+			stopFlag = kEventFlagEscapeKey;
+			break;
+		}
+
+		// TODO: Hot rectangles
+		if ((flags & kEventFlagHotRectangle) /* && event.type == SCI_EVENT_HOT_RECTANGLE */) {
+			warning("Hot rectangles not implemented in VMD player");
+			stopFlag = kEventFlagHotRectangle;
```

Comment:
AFAIK, only Phantasmagoria 1 actually uses the Hot rectangles functionality, and that does not use VMD videos. So, in theory, this functionality is not actually used anywhere, so we could safely error out here